# Softwares used in the biomarker module

## [cnvkit2scarhrd](https://github.com/hydra-genetics/biomarker/blob/develop/workflow/scripts/cnvkit2scarhrd.py)
Script that modifies the cnvkit segmentation file so that it can be used by scarhrd.

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__scarhrd__cnvkit2scarhrd#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__scarhrd__cnvkit2scarhrd#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__cnvkit2scarhrd#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__cnvkit2scarhrd#

---

## [finaletoolkit_end_motifs](https://github.com/epifluidlab/FinaleToolkit/wiki/End-Motifs)
Use the finaletoolkit package to calculate end motif frequencies in plasma samples

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__finaletoolkit__finaletoolkit_end_motifs#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__finaletoolkit__finaletoolkit_end_motifs#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__finaletoolkit_end_motifs#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__finaletoolkit_end_motifs#


## [finaletoolkit_interval_end_motifs](https://github.com/epifluidlab/FinaleToolkit/wiki/End-Motifs#interval-end-motifs)
Use the finaletoolkit package to calculate end motif frequencies for each region in plasma samples

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__finaletoolkit__finaletoolkit_interval_end_motifs#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__finaletoolkit__finaletoolkit_interval_end_motifs#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__finaletoolkit_interval_end_motifs#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__finaletoolkit_interval_end_motifs#


## [finaletoolkit_mds](https://github.com/epifluidlab/FinaleToolkit/wiki/Motif-Diversity-Score)
Use the finaletoolkit package to calculate the motif diversity score in plasma samples. The score is based on end motif frequencies generated by the rule finaletoolkit_end_motifs.

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__finaletoolkit__finaletoolkit_mds#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__finaletoolkit__finaletoolkit_mds#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__finaletoolkit_mds#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__finaletoolkit_mds#


## [finaletoolkit_interval_mds](https://github.com/epifluidlab/FinaleToolkit/wiki/Motif-Diversity-Score#interval-mds)
Use the finaletoolkit package to calculate the motif diversity score for each region in bed file in plasma samples. The score is based on end motif frequencies generated by the rule finaletoolkit_interval_end_motifs.

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__finaletoolkit__finaletoolkit_interval_mds#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__finaletoolkit__finaletoolkit_interval_mds#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__finaletoolkit_interval_mds#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__finaletoolkit_interval_mds#


## [finaletoolkit_frag_length_bins](https://epifluidlab.github.io/FinaleToolkit/documentation/cli_reference/index.html#frag-length-bins)
Calculate histogram with fragment lengths from bam file from a plasma sample

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__finaletoolkit__finaletoolkit_frag_length_bins#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__finaletoolkit__finaletoolkit_frag_length_bins#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__finaletoolkit_frag_length_bins#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__finaletoolkit_frag_length_bins#


---

## [fix_scarhrd_output](https://github.com/hydra-genetics/biomarker/blob/develop/workflow/scripts/fix_scarhrd_output.py)
Script that modifies the scarhrd output for improved readability of the resulting score.

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__scarhrd__fix_scarhrd_output#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__scarhrd__fix_scarhrd_output#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__fix_scarhrd_output#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__fix_scarhrd_output#

---

## [fragmentomics_fragment_length_patient_score](https://github.com/pughlab/fragmentomics/tree/main/fragment_score)
Calculates a fragment length score for each cfDNA sample in order to get an indication of the ctDNA content level.

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__fragmentomics__fragmentomics_fragment_length_patient_score#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__fragmentomics__fragmentomics_fragment_length_patient_score#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__fragmentomics_fragment_length_patient_score#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__fragmentomics_fragment_length_patient_score#

---

## [msisensor_pro](https://github.com/xjtu-omics/msisensor-pro)
Calculates the % of microsatelites detected that are instable which can be used to determine MSS or MSI status.

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__msisensor_pro__msisensor_pro#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__msisensor_pro__msisensor_pro#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__msisensor_pro#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__msisensor_pro#

---

## [msisensor_pro_filter_sites](https://github.com/hydra-genetics/biomarker/blob/develop/workflow/scripts/msisensor_pro_filter_sites.py)
Script that takes the PoN for MSISensorPro and creates a new PoN with only msi sites found in the supplied bedfile. If bedfile is set to empty string no filtering is performed.

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__msisensor_pro__msisensor_pro_filter_sites#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__msisensor_pro__msisensor_pro_filter_sites#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__msisensor_pro_filter_sites#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__msisensor_pro_filter_sites#

---

## [scarhrd](https://github.com/sztup/scarHRD)
Calculates a HRD score by the sum of three different scores that counts the number of cnv events found. The program uses an external segmentation file from for example cnvkit.

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__scarhrd__scarhrd#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__scarhrd__scarhrd#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__scarhrd#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__scarhrd#

---

## [tmb](https://github.com/hydra-genetics/biomarker/blob/develop/workflow/scripts/tmb.py)
Python script that calculates the tumor mutational burden. Filters the vcf file based on vep annotations to obtain a confident set of somatic nsSNVs. The file can optionally be filtered by an artifact and background noise file. 

### :snake: Rule

#SNAKEMAKE_RULE_SOURCE__tmb__tmb#

#### :left_right_arrow: input / output files

#SNAKEMAKE_RULE_TABLE__tmb__tmb#

### :wrench: Configuration

#### Software settings (`config.yaml`)

#CONFIGSCHEMA__tmb#

#### Resources settings (`resources.yaml`)

#RESOURCESSCHEMA__tmb#

